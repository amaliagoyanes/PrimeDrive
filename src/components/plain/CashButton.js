import React, {Component} from 'react';
import {
    TouchableOpacity,
    Text,
    View,
} from 'react-native';
import {Row, Col} from 'react-native-easy-grid';

import {ButtonItem} from '../index'
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling';

export default class CashButton extends Component {
    constructor(props) {
        super(props);
        this.state = {
            amount: '', active: false
        };

        this.activate = this.activate.bind(this);
        this.deactivate = this.deactivate.bind(this);
        this.onPressBtn = this.onPressBtn.bind(this);
        this.getAmount = this.getAmount.bind(this);
        this.setAmount = this.setAmount.bind(this);
    }

    deactivate() {
        this.setState({active: false})
    }

    activate() {
        this.setState({active: true})
    }

    onPressBtn() {
        this.props.activate(this.props.refId, this.state.amount)
    }

    setAmount(amount) {
        this.setState({amount: amount})
    }

    getAmount() {
        return this.state.amount
    }

    render() {
        let bgColor = this.state.active ? {backgroundColor: '#404048'} : {};
        let amount = this.state.amount !== '' ? this.state.amount : undefined;
        return (
            <ButtonItem text={this.props.text} detailsText={amount} onPress={this.onPressBtn}
                        style={[styles.modalButton, styles.openDayButton, styles.buttonSidedText, bgColor,
                            {alignItems: 'center', justifyContent: 'center', height: scale(50),}]}
                        textStyle={[styles.modalButtonText, this.props.style, {
                            fontSize: scaleText(12), textAlign: 'left', width: this.state.amount !== '' ? '50%' : '100%'
                        }]}/>
        )
    }
}