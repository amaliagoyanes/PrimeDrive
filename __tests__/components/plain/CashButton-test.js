'use strict';

import 'react-native';
import React from 'react';
import CashButton from '../../../src/components/plain/CashButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <CashButton text="Test"/>).toJSON();
    expect(tree).toMatchSnapshot();
});
