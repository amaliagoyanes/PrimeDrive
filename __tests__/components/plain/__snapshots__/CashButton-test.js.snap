// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#66666d",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 15,
          "opacity": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Gotham-Book",
              "fontSize": 14,
            },
            Array [
              Object {
                "color": "#fefefe",
                "fontSize": 16,
              },
              Object {
                "fontSize": 12,
                "textAlign": "left",
                "width": "100%",
              },
            ],
          ]
        }
      >
        TEST
      </Text>
    </View>
  </View>
</View>
`;
